File: cookbooks/03-photos-to-product-model/python/main.py
"""Turn three product photos into a life-size, 4K-textured GLB and USDZ."""

import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parents[3]))  # so `shared` imports
from shared.python.meshy import Meshy

PHOTOS = [
    Path("../input/armchair-001/1-front.jpg"),
    Path("../input/armchair-001/2-back.jpg"),
    Path("../input/armchair-001/3-side.jpg"),
]
OUTPUT = Path("output")


def main() -> None:
    client = Meshy()  # reads MESHY_API_KEY from .env
    task_id = client.create(
        "multi-image-to-3d",
        {
            "image_urls": [Meshy.data_uri(p) for p in PHOTOS],
            "ultra_mode": True,
            "should_texture": True,
            "enable_pbr": True,
            "texture_resolution": "4k",
            "auto_size": True,
            "origin_at": "bottom",
            "multi_view_thumbnails": True,
            "target_formats": ["glb", "usdz"],
        },
    )
    task = client.wait("multi-image-to-3d", task_id)
    glb = client.download(task["model_urls"]["glb"], OUTPUT / "armchair.glb")
    client.download(task["model_urls"]["usdz"], OUTPUT / "armchair.usdz")
    for view, url in task["thumbnail_urls"].items():
        client.download(url, OUTPUT / f"armchair-{view}.png")
    print(f"Done: {glb}  ({task['consumed_credits']} credits)")


if __name__ == "__main__":
    main()

File: cookbooks/03-photos-to-product-model/typescript/main.ts
// Turn three product photos into a life-size, 4K-textured GLB and USDZ.

import { Meshy } from "../../../shared/typescript/meshy.js";

const PHOTOS = [
  "../input/armchair-001/1-front.jpg",
  "../input/armchair-001/2-back.jpg",
  "../input/armchair-001/3-side.jpg",
];
const OUTPUT = "output";

const client = new Meshy(); // reads MESHY_API_KEY from .env
const taskId = await client.create("multi-image-to-3d", {
  image_urls: await Promise.all(PHOTOS.map((path) => Meshy.dataUri(path))),
  ultra_mode: true,
  should_texture: true,
  enable_pbr: true,
  texture_resolution: "4k",
  auto_size: true,
  origin_at: "bottom",
  multi_view_thumbnails: true,
  target_formats: ["glb", "usdz"],
});
const task = await client.wait("multi-image-to-3d", taskId);
const glb = await client.download(task.model_urls.glb, `${OUTPUT}/armchair.glb`);
await client.download(task.model_urls.usdz!, `${OUTPUT}/armchair.usdz`);
for (const [view, url] of Object.entries(task.thumbnail_urls!)) {
  await client.download(url, `${OUTPUT}/armchair-${view}.png`);
}
console.log(`Done: ${glb}  (${task.consumed_credits} credits)`);
